interfaces/builtin/repowerd.go

```diff
@@ -0,0 +1,335 @@
+# Description: Allow operating as the repowerd service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# HW alarms
+/dev/alarm rw,
+
+# Suspend
+/sys/power/** rw,
+
+# Backlight
+/sys/**/brightness rw,
+/sys/**/max_brightness r,
+/sys/**/type r,
+
+# DBus
+#include <abstractions/dbus-strict>
+
+dbus (send)
+  bus=system
+  path=/org/freedesktop/DBus
+  interface=org.freedesktop.DBus
+  member={GetId,RequestName,ReleaseName,GetConnectionUnixProcessID}
+  peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+  bus=system
+  interface=org.freedesktop.DBus.{Properties,Introspectable},
+
+dbus (bind)
+  bus=system
+  name=com.canonical.powerd,
+
+dbus (bind)
+  bus=system
+  name=com.canonical.Unity.Screen,
+
+dbus (receive, send)
+  bus=system
+  interface=com.canonical.powerd,
+
+dbus (receive, send)
+  bus=system
+  interface=com.canonical.Unity.Screen,
+
+# logind session
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member=GetSessionByPID
+  peer=(name=org.freedesktop.login1),
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1{,/seat/**,/session/**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.freedesktop.login1),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member={SessionAdded,SessionRemoved}
+  peer=(name=org.freedesktop.login1),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/login1{,/seat/**,/session/**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.freedesktop.login1),
+
+# logind power
+dbus (send)
+  bus=system
+  path=/org/freedesktop/login1
+  interface=org.freedesktop.login1.Manager
+  member={Hibernate,HybridSleep,Inhibit,PowerOff,Suspend}
+  peer=(name=org.freedesktop.login1),
+
+# UPower
+dbus (send)
+  bus=system
+  path=/org/freedesktop/UPower
+  interface=org.freedesktop.UPower
+  member=EnumerateDevices
+  peer=(name=org.freedesktop.UPower),
+dbus (send)
+  bus=system
+  path=/org/freedesktop/UPower{,/devices/**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.freedesktop.UPower),
+dbus (receive)
+  bus=system
+  path=/org/freedesktop/UPower{,/devices/**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.freedesktop.UPower),
+
+# Ofono
+dbus (send)
+  bus=system
+  path=/
+  interface=org.ofono.Manager
+  member=GetModems
+  peer=(name=org.ofono),
+dbus (send)
+  bus=system
+  interface=org.ofono.RadioSettings
+  member=SetProperty
+  peer=(name=org.ofono),
+dbus (send)
+  bus=system
+  path=/{,**}
+  interface=org.freedesktop.DBus.Properties
+  member=Get{,All}
+  peer=(name=org.ofono),
+dbus (receive)
+  bus=system
+  path=/{,**}
+  interface=org.freedesktop.DBus.Properties
+  member=PropertiesChanged
+  peer=(name=org.ofono),
+
+# Unity.Display
```

Comment:
As said above, we're talking about the slot here and not the plug. You're right clients should be able to use just the repowerd plug to talk with repowerd and thats it. But I guess today no client talks with ofono directly and if it does its an implementation detail of that snap and doesn't relate to the use of repowerd.